src/snapred/backend/recipe/algorithm/PixelGroupingParametersCalculationAlgorithm.py

```diff
@@ -114,21 +117,44 @@ def PyExec(self):
 
         outputParams = json.dumps(allGroupingParams_json)
         self.setProperty("OutputParameters", outputParams)
+        self.mantidSnapper.DeleteWorkspace("Cleaning up grouped workspace.", Workspace=grouped_ws_name)
+        self.mantidSnapper.DeleteWorkspace("Cleaning up resolution workspace.", Workspace=pgpca_resolution_ws)
+        self.mantidSnapper.DeleteWorkspace(
+            "Cleaning up partial resolution workspace.", Workspace=pgpca_resolution_partials
+        )
+        self.mantidSnapper.executeQueue()
 
     # create a grouping workspace from a grouping file
     def CreateGroupingWorkspace(self, grouping_ws_name):
         groupingFilePath = self.getProperty("GroupingFile").value
-        file_extension = pathlib.Path(groupingFilePath).suffix
-        if file_extension.upper()[1:] == "XML":
+        if "lite" in groupingFilePath:
+            self.mantidSnapper.CreateWorkspace(
+                "Creating Instrument Definition Workspace ...", OutputWorkspace="idf", DataX=1, DataY=1
+            )
+            self.mantidSnapper.LoadInstrument(
+                "Loading instrument definition file ...",
+                Workspace="idf",
+                Filename="/SNS/SNAP/shared/Calibration/Powder/SNAPLite.xml",
+                MonitorList="-2--1",
+                RewriteSpectraMap=False,
+            )
```

Comment:
A lot of this was taken care of by `CustomGroupWorkspace` but I havent taken the time to fully understand this algo enough to know if this set of alternative steps and path are truely necessary.

Currently the `CustomGroupWorkspace`  sets up one for every group, but this algo is setup to take one group at a time.  It would require a little bit of refactor.